ballerina/modules/serializer/tests/lib_test.bal

```diff
@@ -86,14 +86,43 @@ function testQuotesForInvalidPlanarChar(string line) returns error? {
 
 function invalidPlanarDataGen() returns map<[string]> {
     return {
-        "comment": [" #"],
+        "comment": [" # comment"],
         "explicit key": ["? "],
         "sequence entry": ["- "],
         "mapping value": [": "],
-        "flow indicator": ["}a"]
+        "flow indicator": ["}a"],
+        "alias": ["*/*"],
+        "collect-entry": [", "],
+        "anchor": ["&anchor"],
+        "tag": ["!tag"],
+        "directive": ["%YAML 1.2"]
     };
 }
 
+@test:Config {
+    dataProvider: scalarWithNewLinesDataGen,
+    groups: ["serializer"]
+}
+function testScalarWithNewLines(json line, string[] expectedOutputs) returns error? {
+    common:Event[] events = check getSerializedEvents(line);
+    int index = 0;
+    foreach common:Event event in events {
+        if event is common:ScalarEvent {
```

Comment:
We need to add a `assertFail` scenario for the else condition

Reply:
The test is designed to consider only the ScalarEvents and ignore the rest, as the new lines are only processed in these events. Hence, the events that are not ScalarEvents form a valid stream for the given input.